src/playlist-selectors.js

```diff
@@ -113,81 +113,102 @@ export const comparePlaylistResolution = function(left, right) {
-  let bandwidthBestVariant;
-  let resolutionPlusOne;
-  let resolutionBestVariant;
-  let haveResolution;
-  let resolutionPlusOneList = [];
-  let resolutionPlusOneSmallest = [];
-  let resolutionBestVariantList = [];
-
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+const simpleSelector = function(master, playerBandwidth, playerWidth, playerHeight) {
+  // convert the playlists to an intermediary representation to make comparisons easier
+  let sortedPlaylistReps = master.playlists.map((playlist) => {
+    let width;
+    let height;
+    let bandwidth;
+
+    if (playlist.attributes) {
+      width = playlist.attributes.RESOLUTION && playlist.attributes.RESOLUTION.width;
+      height = playlist.attributes.RESOLUTION && playlist.attributes.RESOLUTION.height;
+      bandwidth = playlist.attributes.BANDWIDTH;
+    }
+
+    bandwidth = bandwidth || window.Number.MAX_VALUE;
+
+    return {
+      bandwidth,
+      width,
+      height,
+      playlist
+    };
+  });
+
+  stableSort(sortedPlaylistReps, (left, right) => left.bandwidth - right.bandwidth);
 
   // filter out any playlists that have been excluded due to
   // incompatible configurations or playback errors
-  sortedPlaylists = sortedPlaylists.filter(Playlist.isEnabled);
+  sortedPlaylistReps = sortedPlaylistReps.filter(
+    (rep) => Playlist.isEnabled(rep.playlist)
+  );
+
   // filter out any variant that has greater effective bitrate
   // than the current estimated bandwidth
-  bandwidthPlaylists = sortedPlaylists.filter(function(elem) {
-    return elem.attributes &&
-           elem.attributes.BANDWIDTH &&
-           elem.attributes.BANDWIDTH * Config.BANDWIDTH_VARIANCE < bandwidth;
-  });
+  let bandwidthPlaylistReps = sortedPlaylistReps.filter(
+    (rep) => rep.bandwidth * Config.BANDWIDTH_VARIANCE < playerBandwidth
+  );
+
+  let highestRemainingBandwidthRep =
+    bandwidthPlaylistReps[bandwidthPlaylistReps.length - 1];
 
   // get all of the renditions with the same (highest) bandwidth
   // and then taking the very first element
-  bandwidthBestVariant = bandwidthPlaylists.filter(function(elem) {
-    return elem.attributes.BANDWIDTH === bandwidthPlaylists[bandwidthPlaylists.length - 1].attributes.BANDWIDTH;
-  })[0];
-
-  // sort variants by resolution
-  stableSort(bandwidthPlaylists, comparePlaylistResolution);
+  let bandwidthBestRep = bandwidthPlaylistReps.filter(
+    (rep) => rep.bandwidth === highestRemainingBandwidthRep.bandwidth
+  )[0];
 
   // filter out playlists without resolution information
-  haveResolution = bandwidthPlaylists.filter(function(elem) {
-    return elem.attributes &&
-           elem.attributes.RESOLUTION &&
-           elem.attributes.RESOLUTION.width &&
-           elem.attributes.RESOLUTION.height;
-  });
+  let haveResolution = bandwidthPlaylistReps.filter((rep) => rep.width && rep.height);
+
+  // sort variants by resolution
+  stableSort(haveResolution, (left, right) => left.width - right.width);
```

Comment:
ONe reason is that we reuse `haveResolution` in different filters to get different values, so it's probably better to sort once then filter twice than filter twice and sort twice. We only ever sort twice, once by bandwidth and once by resolution and I think its easier to reason about the filters when they are working on sorted lists